superset-frontend/src/explore/exploreUtils.js

```diff
@@ -309,19 +310,28 @@ export const useDebouncedEffect = (effect, delay) => {
   }, [callback, delay]);
 };
 
-export const getSimpleSQLExpression = (
-  subject,
-  operator,
-  comparator,
-  isMulti,
-) => {
+export const getSimpleSQLExpression = (subject, operator, comparator) => {
+  const isMulti = MULTI_OPERATORS.has(operator);
   let expression = subject ?? '';
   if (subject && operator) {
     expression += ` ${operator}`;
-    if (comparator) {
-      expression += ` ${isMulti ? "('" : ''}${comparator}${
-        isMulti ? "')" : ''
-      }`;
+    const firstValue =
+      isMulti && Array.isArray(comparator) ? comparator[0] : comparator;
+    const comparatorArray =
+      comparator === undefined ||
+      comparator === null ||
+      Array.isArray(comparator)
+        ? comparator
+        : [comparator];
+    const isString =
+      firstValue !== undefined && Number.isNaN(Number(firstValue));
```

Comment:
Can we replace it with `typeof firstValue === 'string'`?

Reply:
I originally did, but then it hit me that numeric comparators are in fact strings if entered manually, hence need to be cast first.

Reply:
Ah, but that wouldn't handle cases like `"1"`. Nevermind 🙂